### PR TITLE
Add config to disable HLC realtime segment completion

### PR DESCRIPTION
**Pull request by jackjlli**
This PR adds a config to explicitly disable HLC realtime segment completion.
By default it is enabled.

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ private static long getRandomInitialDelayInSeconds() {
   private static final String REALTIME_SEGMENT_METADATA_COMMIT_NUMLOCKS =
       "controller.realtime.segment.metadata.commit.numLocks";
   private static final String ENABLE_STORAGE_QUOTA_CHECK = "controller.enable.storage.quota.check";
-
+  private static final String ENABLE_HLC_REALTIME_SEGMENT_COMPLETION = "controller.enable.hlc.realtime.segment.completion";
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: Can we rename the config as ALLOW_HLC_TABLES? Please document the config clearly. If the config says "true"(default),  then the behavior is unchanged. If the config  is set to "false", then the following will happen:
* Existing  HLC realtime tables will stop consumption
* Creation of mew HLC tables will be disallowed.

**Comment by mcvsubbu** [comment]
Also, please change the LLCRealtimeCluster test to work with this disabled setting

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 2]
```diff
@@ -136,8 +135,8 @@ private static long getRandomInitialDelayInSeconds() {
   private static final String REALTIME_SEGMENT_METADATA_COMMIT_NUMLOCKS =
       "controller.realtime.segment.metadata.commit.numLocks";
   private static final String ENABLE_STORAGE_QUOTA_CHECK = "controller.enable.storage.quota.check";
-
   private static final String ENABLE_BATCH_MESSAGE_MODE = "controller.enable.batch.message.mode";
+  private static final String ALLOW_HLC_TABLES = "controller.allow.hlc.tables";
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: please add a paragraph to describe what happns when this is turned off.

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java [hunk 3]
```diff
@@ -273,8 +273,14 @@ private void setUpPinotController() {
         new SegmentCompletionManager(helixParticipantManager, _pinotLLCRealtimeSegmentManager, _controllerMetrics,
             _controllerLeadershipManager, _config.getSegmentCommitTimeoutSeconds());
 
-    _realtimeSegmentsManager = new PinotRealtimeSegmentManager(_helixResourceManager, _controllerLeadershipManager);
-    _realtimeSegmentsManager.start(_controllerMetrics);
+    if (_config.getHLCTablesAllowed()) {
+      LOGGER.info("HLC table is allowed. HLC realtime segment completion is enabled.");
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: I suggest the log message to be "Realtime tables with High Level consumers will be supported"

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java [hunk 4]
```diff
@@ -273,8 +273,14 @@ private void setUpPinotController() {
...
+      _realtimeSegmentsManager = new PinotRealtimeSegmentManager(_helixResourceManager, _controllerLeadershipManager);
+      _realtimeSegmentsManager.start(_controllerMetrics);
+    } else {
+      LOGGER.info("HLC table is disallowed. HLC realtime segment completion is disabled.");
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: Suggest change the log message to: "Realtime tables with High Level consumers will NOT be supported"

File: pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java [hunk 5]
```diff
@@ -1408,6 +1421,11 @@ public void updateIndexingConfigFor(String tableName, TableType type, IndexingCo
     setExistingTableConfig(tableConfig, tableNameWithType, type);
 
     if (type == TableType.REALTIME) {
+      // Check if HLC table is allowed
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Did you look into the possibility of having a single validateIndexingConfig method?

Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java]: Added.

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java (hunk 4 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: Updated.

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java (hunk 3 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerStarter.java]: Updated.

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 2 above)
Comment (jackjlli) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: Added.

**Pull request merged**